docker/scripts-ci/setup.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+set -o errexit -o nounset
+command -v shellcheck >/dev/null && shellcheck "$0"
+
+PASSWORD=${PASSWORD:-1234567890}
+CHAIN_ID=${CHAIN_ID:-nois-testing}
+MONIKER=${MONIKER:-nois-moniker}
+
+TOKEN=${TOKEN:-unois}
+TRANSFER_PORT=${TRANSFER_PORT:-transfer}
+
+# both types of tokens
+START_BALANCE="1000000000$TOKEN"
+
+echo "Creating genesis ..."
+rm -rf "${HOME}/.noisd"
+noisd init --chain-id "$CHAIN_ID" "$MONIKER"
+cd "${HOME}/.noisd"
+
+sed -i "s/\"stake\"/\"$TOKEN\"/" config/genesis.json # staking/governance token is hardcoded in config, change this
+sed -i "s/\"port_id\": *\"transfer\"/\"port_id\": \"$TRANSFER_PORT\"/" config/genesis.json # allow custom ibc transfer port
+
+# this is essential for sub-1s block times (or header times go crazy)
+sed -i 's/"time_iota_ms": "1000"/"time_iota_ms": "10"/' config/genesis.json
+
+echo "Setting up validator ..."
+if ! noisd keys show validator 2>/dev/null; then
+  echo "Validator does not yet exist. Creating it ..."
+  (
+    echo "$PASSWORD"
+    echo "$PASSWORD"
+  ) | noisd keys add validator
+fi
+# hardcode the validator account for this instance
+echo "$PASSWORD" | noisd add-genesis-account validator "$START_BALANCE"
+
+echo "Setting up accounts ..."
+# (optionally) add a few more genesis accounts
+for addr in "$@"; do
+  echo "$addr"
+  noisd add-genesis-account "$addr" "$START_BALANCE"
+done
+
+echo "Creating genesis tx ..."
+SELF_DELEGATION="3000000$TOKEN" # 3 NOIS (leads to a voting power of 3)
+(
+  echo "$PASSWORD"
+  echo "$PASSWORD"
+  echo "$PASSWORD"
+) | noisd gentx validator "$SELF_DELEGATION" --offline --chain-id "$CHAIN_ID" --moniker="$MONIKER"
+
+ls -lA config/gentx/
+
+echo "Collecting genesis tx ..."
+noisd collect-gentxs
+
+# so weird, but found I needed the -M flag after lots of debugging odd error messages
+# happening when redirecting stdout
+jq -S -M . < config/genesis.json > genesis.tmp
+mv genesis.tmp config/genesis.json
+chmod a+rx config/genesis.json
+
+# Custom settings in config.toml
+sed -i"" \
+  -e 's/^cors_allowed_origins =.*$/cors_allowed_origins = ["*"]/' \
+  -e 's/^timeout_propose =.*$/timeout_propose = "100ms"/' \
+  -e 's/^timeout_propose_delta =.*$/timeout_propose_delta = "100ms"/' \
+  -e 's/^timeout_prevote =.*$/timeout_prevote = "100ms"/' \
+  -e 's/^timeout_prevote_delta =.*$/timeout_prevote_delta = "100ms"/' \
+  -e 's/^timeout_precommit =.*$/timeout_precommit = "100ms"/' \
+  -e 's/^timeout_precommit_delta =.*$/timeout_precommit_delta = "100ms"/' \
+  -e 's/^timeout_commit =.*$/timeout_commit = "200ms"/' \
+  "config/config.toml"
```

Comment:
are these really needed? I made some defaults here https://github.com/noislabs/noisd/blob/main/cmd/noisd/init.go#L111-L117

`starsd init` will start with those settings, although no harm on setting them.

Reply:
The values here are for the CI setting. It has a block time of 250ms instead of 2.5s. This file is only used for the special noisd-ci image.

Reply:
makes sense